src/query_router.rs

```diff
@@ -215,6 +216,50 @@ impl QueryRouter {
         // server it'll go to if the query parser is enabled.
         if matches.len() != 1 {
             debug!("Regular query, not a command");
+            if comment_shard_routing_enabled {
+                // comment_shard_routing_enabled is true, but no sharding comment was found
+                match self.pool_settings.no_shard_specified_behavior {
+                    Some(behavior) => {
+                        match behavior {
+                            NoShardSpecifiedHandling::Random => {
+                                self.set_shard(rand::random::<usize>() % self.pool_settings.shards);
+                            }
+
+                            NoShardSpecifiedHandling::RandomHealthy => {
+                                let pool = get_pool(
+                                    &self.pool_settings.db,
+                                    &self.pool_settings.user.username,
+                                );
+                                match pool {
+                                    Some(pool) => {
+                                        self.set_shard(
+                                            //pool.get_random_shard_id_with_least_errors(),
+                                            pool.get_random_healthy_shard_id(),
+                                        );
+                                    }
+                                    None => {
+                                        // Query is using an obselete pool (old pool from before config reload)
+                                        // so we fallback to just random
+                                        self.set_shard(
+                                            rand::random::<usize>() % self.pool_settings.shards,
+                                        );
+                                    }
+                                }
+                            }
+
+                            NoShardSpecifiedHandling::Shard(shard_id) => {
+                                if shard_id >= self.pool_settings.shards {
+                                    warn!("Shard id {} is out of range, defaulting to 0", shard_id);
+                                    self.set_shard(0);
```

Comment:
Does this represent a significant functionality change? If there's 5 shards and someone queries shard 6 the current version triggers an error and this will now reroute it to shard 0 with no indication of that to the client. Am I reading this code right?

Reply:
Good point. I can make that change. 
Does the current implementation trigger an error or does it panic?

Reply:
Current version panics. It would be far better if it returned an error to the client, but also that logic is likely better somewhere else and in another PR.